application/controllers/Admin.php

```diff
@@ -55,6 +56,87 @@ public function markAsPaid() {
         }
     }
 
+    /**
+     * Checks an input key against a key stored in a file. If it matches, store a cookie on the users browser.
+     */
+    public function authenticate() {
+
+        //gets the passkey from private_keys/auth_props.json
+        if (!defined('Passkey')) {
+            $json = file_get_contents("private_keys/auth_props.json");
+            $authProps = json_decode($json, true);
+        
+            define('Passkey', $authProps["auth.passkey"]);
+        }
+        // Scrambler key
+        $jwtKey = 'JWTKEY';
```

Comment:
A few things with this:
- constants should always be defined as uppercase letters, so change `Passkey` to be like `ADMIN_AUTH_PASSKEY`
- We can move this whole block into `config/constants.php`
- We should also set the scrambler key in `auth_props.json` instead of in the code itself since this is also a very important secret we have to store securely. This can be defined in the same way as Passkey, so maybe a name like `ADMIN_AUTH_PRIVATE_KEY`